search/ranking_utils.cpp

```diff
@@ -1,21 +1,60 @@
 #include "search/ranking_utils.hpp"
+#include "search/token_slice.hpp"
+#include "search/utils.hpp"
 
 #include "base/dfa_helpers.hpp"
 
+#include "indexer/categories_holder.hpp"
+#include "indexer/feature_data.hpp"
+
 #include "std/transform_iterator.hpp"
 
 #include <algorithm>
 #include <sstream>
 
+using namespace std;
 using namespace strings;
 
 namespace search
 {
 namespace
 {
+struct TokenInfo
+{
+  bool m_isCategoryToken = false;
+  bool m_inFeatureTypes = false;
+};
+
 UniString AsciiToUniString(char const * s) { return UniString(s, s + strlen(s)); }
 }  // namespace
 
+// CategoriesInfo ----------------------------------------------------------------------------------
```

Comment:
What about a separating comment for standalone functions?

Reply:
Done